R/read.R

```diff
@@ -854,10 +872,22 @@ formatTxSpots <- function(file, dest = c("rowGeometry", "colGeometry"),
     mols <- mols[mols[[phred_col]] >= min_phred,]
   }
   message(">>> Converting transcript spots to geometry")
+  if (dest == "colGeometry") {
+      if (!length(cell_col) || any(!cell_col %in% names(mols)))
+          stop("Column indicating cell ID not found.")
+      mols <- mols[mols[[cell_col[1]]] != not_in_cell_id,]
+      if (length(cell_col) > 1L) {
+          if (!is.data.table(mols)) ..cell_col <- cell_col
+          cell_col_use <- do.call(paste, c(mols[,..cell_col], sep = "_"))
+          mols$cell_id_ <- cell_col_use
+          mols[,cell_col] <- NULL
+          cell_col <- "cell_id_"
+      }
+  }
   if (z_option == "split" && use_z) {
       mols <- split(mols, mols[[spatialCoordsNames[3]]])
       mols <- lapply(mols, .mols2geo_split, dest = dest,
-                     spatialCoordsNames = spatialCoordsNames,
+                     spatialCoordsNames = spatialCoordsNames[1:2],
```

Comment:
the issue seems to be here, `spatialCoordsNames = spatialCoordsNames` will make `XYZ` MULTIPOINT

Reply:
That's because now by default a separate XY geometry is made for each z-plane. You can use `z_option = "3d"` to not to split z-planes. The reason why is that geometric operations won't work for 3D geometries, and that it takes more data wrangling to plot the transcript spots when we have XYZ geometries.

Reply:
I see, sounds good 👍